### PR TITLE
Abort deletion of primary entity if deletion of dependent entity fails.

### LINKED ISSUE #14434: Status of delete() of associated model is ignored when  cascadeCallback is true.
This is a (multiple allowed):

* [x] bug
* [x] enhancement

* CakePHP Version: 3.8.5
* Platform and Target: Linux, Apache, Mysql.

When associated model is set to be dependent and cascadeCallbacks is set to true, cakephp triggers separate delete fro each associated entity. If this delete is prohibited by application rules, this information is silently ignored (since return status of delete is not checked). This results in DB corruption when parent entity is deleted, and associated records are still present in DB without any errors being detected.

### PR DESCRIPTION
**Pull request by ADmad**
Fixes #14434

Would be great if someone can help with tests for each association type :slightly_smiling_face:.

### CODE REVIEW

**Comment by markstory** [comment]
I can help with tests.

**Comment by markstory** [comment]
@ADmad I added tests for the the association methods. Are there other tests you wanted to see?

**Comment by ADmad** [comment]
Thanks @markstory. I added a additional integration for `Table::delete()` too showing that deletion of primary record is also aborted if deleting dependent records fails.

File: tests/TestCase/ORM/TableTest.php [hunk 1]
```diff
@@ -3025,9 +3025,38 @@ public function testDeleteDependentHasMany()
...
+        $entity = $table->get(3);
+        $result = $table->delete($entity);
+        $this->assertFalse($result);
+
+        $query = $articles->find('all', [
+            'conditions' => [
+                'author_id' => $entity->id,
+            ],
+        ]);
+        $this->assertFalse($query->all()->isEmpty(), 'Should find any rows.');
```
Comment (markstory) [inline tests/TestCase/ORM/TableTest.php:3059]:
```suggestion
        $this->assertFalse($query->all()->isEmpty(), 'Should find some rows.');
```

**Pull request merged**